cookies/views/resource.py

```diff
@@ -939,3 +939,12 @@ def create_dataset(request):
         'collection_count': collections.count(),
     }
     return render(request, 'create_dataset.html', context)
+
+
+@login_required
+def delete_resource(request, resource_id):
+    # Extra check to ensure that only resources which do not have primary id get deleted.
+    resource_container = ResourceContainer.objects.get(pk=resource_id)
+    if resource_container.primary_id is None and request.user.id == resource_container.created_by_id:
```

Comment:
you need to check with the permissions here. other people can have access to a resource and can delete them.